packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -254,14 +267,76 @@ class UserSessionManager implements IUserSessionManager {
    * Handles error
    * @param error The error object
    */
-  onError(error: unknown): void {
+  onError(error: unknown, customMessage?: string): void {
     if (this.errorHandler) {
-      this.errorHandler.onError(error, USER_SESSION_MANAGER);
+      this.errorHandler.onError(error, USER_SESSION_MANAGER, customMessage);
     } else {
       throw error;
     }
   }
 
+  /**
+   * A function to make an external request to set the cookie from server side
+   * @param key       cookie name
+   * @param value     encrypted cookie value
+   */
+  setServerSideCookie(cookieData: CookieData[], store: IStore, removeCookie?: boolean): void {
+    try {
+      let encryptedCookieData: EncryptedCookieData[] = [];
+      cookieData.forEach(e => {
+        encryptedCookieData.push({
+          name: e.name,
+          value: removeCookie
+            ? (e.value as string)
+            : store?.encrypt(stringifyWithoutCircular(e.value, false, [], this.logger)),
+        });
+      });
+      encryptedCookieData = encryptedCookieData.filter(e => isDefinedAndNotNull(e.value));
+      if (encryptedCookieData.length > 0 || removeCookie) {
+        this.httpClient?.getAsyncData({
+          url: `${state.serverCookies.dataServerUrl.value}/rsaRequest`,
+          options: {
+            method: 'POST',
+            data: stringifyWithoutCircular({
+              reqType: 'setCookies',
+              workspaceId: state.source.value?.workspaceId,
+              data: {
+                options: {
+                  maxAge: state.storage.cookie.value?.maxage,
+                  path: state.storage.cookie.value?.path,
+                  domain: state.storage.cookie.value?.domain,
+                  sameSite: state.storage.cookie.value?.samesite,
+                  secure: state.storage.cookie.value?.secure,
+                  expires: state.storage.cookie.value?.expires,
+                },
+                cookies: encryptedCookieData,
+              },
+            }) as string,
+            sendRawData: true,
+          },
+          isRawResponse: true,
+          callback: (res, details) => {
+            if (details?.xhr?.status === 200) {
+              cookieData.forEach(each => {
+                const cookieValue = store?.get(each.name);
+                if (cookieValue !== each.value) {
+                  this.logger?.error(FAILED_SETTING_COOKIE_FROM_SERVER_ERROR(each.name));
+                }
+              });
+            } else {
+              this.logger?.error(DATA_SERVER_REQUEST_FAIL_ERROR(details?.xhr?.status));
+              cookieData.forEach(each => {
+                store?.set(each.name, each.value);
+              });
+            }
+          },
+        });
+      }
+    } catch (e) {
+      this.onError(e, FAILED_SETTING_COOKIE_FROM_SERVER_GLOBAL_ERROR);
```

Comment:
We should fallback to the client-side storage option in this case apart from logging the error.

Reply:
Addressed